### ISSUE TITLE
Any chance of merging my derived library

### ISSUE THREAD

**Issue by prog110** [voliva/angular2-interceptors#29]
I have created a derived library from this project long back with quite a lot of additions to this library. I believe its feature complete (support for oauth2 refresh token flows, handling errors, unsubscription, ability to generate mock response/shortcuit, e.t.c)

Here is the link to the library.
https://github.com/1tontech/x-ng4-http-interceptor-dontuse

& corresponding demo for all almost all features is here.

https://github.com/1tontech/x-ng4-http-interceptor-dontuse-demo

However, I don't have time to maintain any bugs that might arise.

I wanted to check with you if you would be interested in merging the code into this project :)

Please let me know your opinion on this?

**Comment by thekalinga** [comment]
I would be happy if you can merge the code selectively into this library aswell (as mine was derived from this library)

**Comment by TheCodeDestroyer** [comment]
@thekalinga Do you mind if I use your npm until this is merged?
It does work with ng4, right?

**Comment by prog110** [comment]
@TheCodeDestroyer Yes. Please check the demo before your try it out. No warranties however :).

If there are issues (which I hope not, as demo addresses most of the scenarios), I might not fix it unless its a critical bug, as I am quite occupied with my own work (which is why I created this ticket to begin with)
